src/aooptionsdialog.cpp

```diff
@@ -934,10 +955,47 @@ void AOOptionsDialog::save_pressed()
     ao_app->initBASS();
 
   callwordsini->close();
+
+  // We most probably pressed "Restore defaults" at some point. Since we're saving our settings, remove the temporary file.
+  if (QFile::exists(ao_app->get_base_path() + "config.temp"))
+      QFile::remove(ao_app->get_base_path() + "config.temp");
   done(0);
 }
 
-void AOOptionsDialog::discard_pressed() { done(0); }
+void AOOptionsDialog::discard_pressed() {
+    // The .temp file exists, meaning we are trying to undo the user clicking on "Restore defaults" and bring back the old settings.
+    if (QFile::exists(ao_app->get_base_path() + "config.temp")) {
+        // Delete the QSettings object so it does not interfere with the file
+        delete ao_app->configini;
+        // Remove the current config.ini
+        QFile::remove(ao_app->get_base_path() + "config.ini");
+        // Rename .temp to .ini
+        QFile::rename(ao_app->get_base_path() + "config.temp", ao_app->get_base_path() + "config.ini");
+        // Recreate the QSettings object from the ini file, restoring the settings before the Options Dialog was opened..
+        ao_app->configini =
+            new QSettings(ao_app->get_base_path() + "config.ini", QSettings::IniFormat);
+    }
+    done(0);
+}
+
+void AOOptionsDialog::button_clicked(QAbstractButton *button) {
+    if (ui_settings_buttons->buttonRole(button) == QDialogButtonBox::ResetRole) {
+        // Store the current settings as a .temp file
+        QFile::rename(ao_app->get_base_path() + "config.ini", ao_app->get_base_path() + "config.temp");
+        // Load up the default settings
+        ao_app->configini->clear();
+        // Update the values on the settings ui
+        update_values();
```

Comment:
Sounds complicated. You just want to read an empty file for QSettings, right? Perhaps read `config.temp`, update values, and delete the file right away.

Reply:
That's what I tried but QSettings behavior overwrote the file lol

Reply:
No but like delete the qsettings instance right after reading.